### PR TITLE
Format top-level and local variable declarations.

### PR DESCRIPTION
**Pull request by munificent**
This required tweaking the way partial solutions are ordered so that the solver doesn't avoid overflowing solutions that will ultimately lead to non-overflowing lower cost ones.

### CODE REVIEW

File: lib/src/piece/variable.dart [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+/// VeryLongTypeName
+/// longVariableName = initializer,
+/// anotherVariable = anotherInitializer;
+/// ```
+class VariablePiece extends Piece {
+  /// Split between each variable in a multiple variable declaration.
+  static const State _betweenVariables = State(1);
+
+  /// Split after the type annotation and between each variable.
+  static const State _afterType = State(2);
```
Comment (natebosch) [inline lib/src/piece/variable.dart:42]: [optional]

```suggestion
  static const _betweenVariables = State(1);

  /// Split after the type annotation and between each variable.
  static const _afterType = State(2);
```

**Review by natebosch — APPROVED** [review]
The commit message could use much more detail.

File: lib/src/piece/variable.dart (hunk 1 above)
Comment (munificent) [reply lib/src/piece/variable.dart:42]: I'm not sure if it's habit or an actually good idea, but my consistent practice is to always type annotate members. <shrug>

Comment (natebosch) [reply lib/src/piece/variable.dart:42]: > consistent practice

https://github.com/dart-lang/dart_style/pull/1279/files#diff-dc088a2965c2b639564656129400eaad08b4e1c424d03bd272d941431146aaceR85-R91

😝

**Pull request merged**